### DISCUSSION TITLE
xmake 使用distcc 编译 提示 attempt to perform arithmetic on a nil value

### DISCUSSION THREAD

**Discussion by skyshkh** [xmake-io/xmake#5576] [General]
xmake service --connect --distcc
: connect 127.0.0.1:9693 ..
: connect 10.0.11.13:9693 ..
: 127.0.0.1:9693 connected!
: 10.0.11.13:9693 connected!
shikh@js-shikaihua:~/work/panda/panda-server$ xmake
error: @programdir/modules/private/service/distcc_build/client.lua:178: attempt to perform arithmetic on a nil value (field 'njob')
stack traceback:
@programdir/core/base/utils.lua:285: in metamethod 'add'
@programdir/modules/private/service/distcc_build/client.lua:178: in method 'maxjobs'
@programdir/modules/private/service/distcc_build/client.lua:187: in method 'freejobs'
@programdir/modules/async/runjobs.lua:184: in function <@programdir/modules/async/runjobs.lua:179>
[C]: in function 'xpcall'
@programdir/core/base/utils.lua:275: in function 'sandbox/modules/utils.trycall'
@programdir/core/base/scheduler.lua:625: in function 'base/scheduler.co_group_begin'
...mdir/core/sandbox/modules/import/core/base/scheduler.lua:110: in field 'co_group_begin'
@programdir/modules/async/runjobs.lua:179: in function <@programdir/modules/async/runjobs.lua:56>
(...tail calls...)
@programdir/actions/build/build.lua:315: in function <@programdir/actions/build/build.lua:303>
(...tail calls...)
@programdir/actions/build/main.lua:109: in global '_do_build'
@programdir/actions/build/main.lua:126: in function <@programdir/actions/build/main.lua:120>
[C]: in function 'xpcall'
@programdir/core/base/utils.lua:275: in function 'sandbox/modules/utils.trycall'
@programdir/core/sandbox/modules/try.lua:117: in global 'try'
@programdir/actions/build/main.lua:118: in global 'build_targets'
@programdir/actions/build/main.lua:201: in function <@programdir/actions/build/main.lua:165>
(...tail calls...)
[C]: in function 'xpcall'
@programdir/core/base/utils.lua:275: in function 'sandbox/modules/utils.trycall'
(...tail calls...)
@programdir/core/base/task.lua:491: in function 'base/task.run'
@programdir/core/main.lua:327: in upvalue 'cotask'
@programdir/core/base/scheduler.lua:406: in function <@programdir/core/base/scheduler.lua:399>

**Comment by waruqi** [comment]
贴下 .xmake/.../status.txt 的内容，我看下。

跟这个问题类似，https://github.com/xmake-io/xmake/issues/4760

也可以按照里面的调试方式调下。

**Reply by waruqi** [reply]
再试下 `xmake update -s dev`
